src/main/java/com/cleanroommc/groovyscript/compat/mods/tinkersconstruct/recipe/MeltingRecipeBuilder.java

```diff
@@ -0,0 +1,63 @@
+package com.cleanroommc.groovyscript.compat.mods.tinkersconstruct.recipe;
+
+import com.cleanroommc.groovyscript.api.GroovyLog;
+import com.cleanroommc.groovyscript.api.IIngredient;
+import com.cleanroommc.groovyscript.core.mixin.tconstruct.MeltingRecipeAccessor;
+import com.cleanroommc.groovyscript.helper.ingredient.OreDictIngredient;
+import com.cleanroommc.groovyscript.helper.recipe.AbstractRecipeBuilder;
+import org.jetbrains.annotations.Nullable;
+import slimeknights.mantle.util.RecipeMatch;
+import slimeknights.tconstruct.library.smeltery.MeltingRecipe;
+
+public class MeltingRecipeBuilder extends AbstractRecipeBuilder<MeltingRecipe> {
+    private int temp = 300;
+    private final MeltingRecipeRegistry registry;
+    private final String recipeName;
+
+    public static RecipeMatch recipeMatchFromIngredient(IIngredient ingredient, int amount) {
+        return (ingredient instanceof OreDictIngredient) ? RecipeMatch.of(((OreDictIngredient) ingredient).getOreDict(), amount) : RecipeMatch.of(ingredient.getMatchingStacks()[0], amount);
+    }
+
+    public static RecipeMatch recipeMatchFromIngredient(IIngredient ingredient) {
+        return recipeMatchFromIngredient(ingredient, 1);
+    }
+
+    public MeltingRecipeBuilder(MeltingRecipeRegistry registry, String recipeName) {
+        this.registry = registry;
+        this.recipeName = recipeName;
+    }
+
+    public MeltingRecipeBuilder temperature(int temp) {
+        this.temp = temp + 300;
```

Comment:
Is this supposed to be a celsius to kelvin conversion? Yeet!

Reply:
This is because Tinkers subtracts 300 from it's temperature requirements, so a temperature of 300 is actually 0 degrees, it should probably be assumed that when a user inputs a temperature of 500, they mean 500 degrees and not 200 degrees.